Capstone/Assets/Game/Scripts/EnemySwing.cs

```diff
@@ -0,0 +1,181 @@
+
+            // When the object should fade out
+            if (fadeOutTimer >= fadeOutLength)
+            {
+                // Shrink the cubes
+                enemyObject.transform.localScale = new Vector3(
+                    enemyObject.transform.localScale.x - (shrinkSpeed * Time.deltaTime), 
+                    enemyObject.transform.localScale.y - (shrinkSpeed * Time.deltaTime), 
+                    enemyObject.transform.localScale.z - (shrinkSpeed * Time.deltaTime));
+
+                // After the cubes are shrunk, destroy it
+                if (enemyObject.transform.localScale.x <= 0 && enemyObject.transform.localScale.y <= 0 && enemyObject.transform.localScale.z <= 0)
+                {
+                    Destroy(enemyObject);
+                }
+
+            }
+
+            // Testing
+            // If the enemy is currently damaging an object
+            if (damaging)
+            {
+                // Update the timer
+                damagingTimer += Time.deltaTime;
+
+                // If the timer is equal to or exceeds the limit
+                if (damagingTimer >= damagingTimerLimit)
+                {
+                    // If the damaging finished event has subcribing delegates
+                    // Call it, running all subscribing delegates
+                    if (DamagingFinished != null)
+                    {
+                        DamagingFinished(ID);
+                    }
+                    // If the damaging finished event doesn't have any subscribing events
+                    // Something has gone wrong because damaging shouldn't be true otherwise
+                    else
+                    {
+                        Debug.Log("Damaging Finished Event was not subscribed to correctly");
+                    }
+
+                    // Reset fields
+                    damagingTimer = 0.0f;
+                    damaging = false;
+                }
+            }
+
+        }
+
+       /// <summary>
+       /// Allows for the player to hit and be hit by the Swinging Enemy
+       /// </summary>
+       /// <param name="other"></param>
+        private void OnTriggerEnter(Collider other)
+        {
+            if(other.GetComponent<Player>() != null || other.GetComponent<Sword>() != null)
+            { 
+                // Checks to see if the player is in the blocking state
+                if (player.GetPlayerFSMState().ID == Enums.PlayerCondition.F_Blocking && other.GetComponent<Sword>() == true)
+                {
+                    // Damage Enemy
+                    isBroken = true;
+                }
+                // Otherwise deal damage to the player
+                else if (other.GetComponent<Player>() == true)
+                {
+                    // Damage Player
+                    //other.GetComponent<Player>().TakeDamage(ID, 1);
+                    ObjectController.DamageableObjects[other.GetComponent<Player>().ID].DamageableObject.TakeDamage(ID, 1);
```

Comment:
Can we do `ObjectController.DamageableObjects[player.ID].DamageableObject.TakeDamage(ID, 1);` instead?